### ISSUE TITLE
Pix2Pix not working as expected for very slight differences in images

### ISSUE THREAD

**Issue by mabdullahrafique** [junyanz/pytorch-CycleGAN-and-pix2pix#465]
Hello Sir ! First of all thanks for this great implementation.
I am working on image to image translation, but in my problem the two domains have very slight differences like input image is men face and output is exact same face with change of lens color ( as example ) which make very less pixels. I have tried pix2pix model it is just recreating the input image without noticing the difference between lens color. I am using U-net for 512x512 images. I have also used FusionNet and it is also not working well. Please tell me how I can tweak Pix2Pix for this task. I have also used L2 Loss but it is also not working.

**Comment by junyanz** [comment]
Maybe your L1 loss is too big. You can try to use a smaller L1 loss.

**Comment by mabdullahrafique** [comment]
To address that issue, I have trained my model to learn how to map the same image to same image. Once it is trained I used its weights to perform my operation so L1 loss is low ( as the entire image is same except some pixels ), but in this way it still does not look at small differences in images.

**Comment by gvenkat21** [comment]
> Maybe your L1 loss is too big. You can try to use a smaller L1 loss.

Hi,

Could you elaborate on that pls?

**Comment by junyanz** [comment]
I mean that one can reduce the weight for L1 loss through `--lambda_L1`. This will encourage the model to focus on small details.

**Comment by youyuge34** [comment]
@mabdullahrafique I encountered the same problem as your's. Can u tackled the problem by reducing 
 the weight of L1 loss ? And which value of weight u choosed?
 THX!

**Comment by EmanueleGhelfi** [comment]
Hi @junyanz I thought the L1 loss was indeed responsible for small details since it's a pixel-wise loss. I thought the GAN loss was responsible for the global structure of the generated images. Do you have any reference?

**Comment by junyanz** [comment]
I think L1 tries to minimize the distortion of the output w.r.t ground truth. GAN loss tries to increase perceptual realism. This nice paper explains it in more depth: https://arxiv.org/abs/1711.06077
